vncviewer/vncviewer.cxx

```diff
@@ -144,18 +144,36 @@ void about_vncviewer()
   fl_message("%s", about_text());
 }
 
-void run_mainloop()
+static int run_mainloop(const char* vncserver, network::Socket* sock)
 {
-  int next_timer;
+  if(exitError != NULL) {
+    free(exitError);
+    exitError = NULL;
+  }
```

Comment:
This looks a bit like it's trying to hide errors. I think it's clearer if you reset `exitError` after you've run `fl_choice()` and the user has asked to reconnect.